// DIAGNOSTICS: -UNUSED_PARAMETER
open class B(x: Int)
class <!CONFLICTING_OVERLOADS!>A<!> : <!SUPERTYPE_INITIALIZED_WITHOUT_PRIMARY_CONSTRUCTOR!>B<!>(1) {
    <!CONFLICTING_OVERLOADS!>constructor()<!>: super(1)
}
